src/test/java/com/ot/restaurant/entities/IoCContainerTest.java

```diff
@@ -0,0 +1,105 @@
+import com.ot.restaurant.exceptions.DependencyValueNullpointerException;
+import com.ot.restaurant.exceptions.DuplicatedDependencyFoundException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class IoCContainerTest {
+
+  @Test
+  void whenInstantiatingIoCContainerShouldInstanceTheSame() {
+
+    IoCContainer ioc = IoCContainer.getInstance();
+    IoCContainer ioc2 = IoCContainer.getInstance();
+
+    assertEquals(ioc, ioc2);
+  }
+
+  @Test
+  void whenRegisterValueShouldResolveValueThenAssertSucceeds() throws Exception {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    String key = "keyEntiti";
+    Object value = new Object();
+
+    ioc.register(key, value);
+
+    Object valueResolvedIoC = ioc.resolve(key);
+
+    assertEquals(value, valueResolvedIoC);
+  }
+
+  @Test
+  public void whenRegisterThrowsDuplicatedExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    String key = "keyEntiti";
+    Object value = new Object();
+
+    com.ot.restaurant.exceptions.DuplicatedDependencyFoundException thrown =
+        Assertions.assertThrows(
+            DuplicatedDependencyFoundException.class,
+            () -> {
+              ioc.register(key, value);
+              ioc.register(key, value);
+            });
+
+    Assertions.assertEquals("Dependency already exists", thrown.getMessage());
+  }
+
+  @Test
+  void whenRegisterThrowsNullExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    Object value = new Object();
+    //String key = "Ramon";
+
+    com.ot.restaurant.exceptions.DependencyKeyNullpointerException thrown =
+        Assertions.assertThrows(
+            DependencyKeyNullpointerException.class,
+            () -> {
+              String key = null;
+              ioc.register(key, value);
+            });
+
+    Assertions.assertEquals("Key Cannot be null", thrown.getMessage());
+  }
+
+  @Test
+  void whenResolveThrowsValueNullpointerExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    com.ot.restaurant.exceptions.DependencyValueNullpointerException thrown =
+        Assertions.assertThrows(
+            DependencyValueNullpointerException.class,
+            () -> {
+              String key = null;
+              ioc.resolve(key);
+            });
+
+    Assertions.assertEquals("Value cannot be null", thrown.getMessage());
+  }
+
+  @Test
+  void whenResolveThrowsNotFoundExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    Object value1 = new Object();
+
+    com.ot.restaurant.exceptions.DependencyNotFoundException thrown =
+        Assertions.assertThrows(
+            DependencyNotFoundException.class,
+            () -> {
+              String key = "keyEntiti1";
+
+              ioc.resolve(key);
+            });
+
+    Assertions.assertEquals("Dependency Value does not exists", thrown.getMessage());
```

Comment:
You can remove this line. Asseting the exception thown is `DependencyNotFoundException` should be enough

Reply:
Done